src/TestSupport/TestLogger.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+namespace Namshi\Cuzzle\TestSupport;
+
+use Psr\Log\AbstractLogger;
+
+/**
+ * Used for testing purposes.
+ *
+ * It records all records and gives you access to them for verification.
+ *
+ */
+class TestLogger extends AbstractLogger
+{
+    /**
+     * @var array
+     */
+    public $records = [];
+
+    public $recordsByLevel = [];
+
+    /**
+     * @inheritdoc
+     */
+    public function log($level, $message, array $context = []): void
+    {
+        $record = [
+            'level' => $level,
+            'message' => $message,
+            'context' => $context,
+        ];
+
+        $this->recordsByLevel[$record['level']][] = $record;
+        $this->records[] = $record;
+    }
+
+    public function hasRecordThatContains($message, $level)
```

Comment:
Should have add some type hints for this method signature?

Reply:
This was a hack to port a small component of the package that made psr 1.x not available. It's purely a test support file so the type hints aren't critical. If you'd like to, will leave with you.